### PR TITLE
Fixed NPE when MDC contains null values.

### PR DESCRIPTION
**Pull request by donbeave**
This is a fix for this issue: https://github.com/getsentry/sentry-java/issues/1361.

### CODE REVIEW

**Comment by marandaneto** [comment]
@donbeave thanks for contributing.
run `make format` and commit the changes, also add an entry to the changelog, thanks.

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 * Fix: set "java" platform to transactions #1332
 * Feat: Add OkHttp client application interceptor (#1330)
 * Fix: Allow disabling tracing through SentryOptions (#1337)
+* Fix: Fix NPE when MDC contains null values (#1364)
```
Comment (marandaneto) [inline CHANGELOG.md:18]:
```suggestion
```

Comment (marandaneto) [reply CHANGELOG.md:18]: @donbeave it should be under `Unreleased`, could you move it? :)

Comment (donbeave) [reply CHANGELOG.md]: @marandaneto Yeah. It's funny how I didn't see it there before. )

Comment (donbeave) [reply CHANGELOG.md]: Fixed.

**Review by marandaneto — APPROVED** [review]

File: sentry-logback/src/main/java/io/sentry/logback/SentryAppender.java [hunk 2]
```diff
@@ -91,8 +90,11 @@ protected void append(@NotNull ILoggingEvent eventObject) {
       event.setExtra("thread_name", loggingEvent.getThreadName());
     }
 
+    // remove keys with null values, there is no sense to send these keys to Sentry
     final Map<String, String> mdcProperties =
-        CollectionUtils.shallowCopy(loggingEvent.getMDCPropertyMap());
+        loggingEvent.getMDCPropertyMap().entrySet().stream()
```
Comment (marandaneto) [inline sentry-logback/src/main/java/io/sentry/logback/SentryAppender.java:95]: mind doing the same fix for log4j2 SentryAdapter? it's pretty much the same thing.

**Pull request merged**